qdrant_client/local/payload_value_extractor.py

```diff
@@ -115,3 +162,77 @@ def test_value_by_key() -> None:
     assert value_by_key(payload, "nested[].empty", flat=False) == [[], [], None]
     assert value_by_key(payload, "the_null", flat=False) == [None]
     # endregion
+
+
+def test_set_value_by_key() -> None:
```

Comment:
Could you please add more tests with "unmatching" paths?

E.g. the path assumes dict key, but the data is an array (`set_value_by_key({"a": []}, {}, "a.b[0]")`), and so on.